share/github-backup-utils/ghe-backup-repositories

```diff
@@ -366,25 +366,8 @@ bm_end "$(basename $0) - Special Data Directories Sync"
 
 if [ -z "$GHE_SKIP_ROUTE_VERIFICATION" ]; then
   bm_start "$(basename $0) - Verifying Routes"
-
-  # The list of gists returned by the source changed in 2.16.23, 2.17.14, 2.18.8 & 2.19.3
-  # so we need to account for this difference here.
-  parse_paths() {
-    while read -r line; do
-      if [[ "$GHE_REMOTE_VERSION" =~ 2.16 && "$(version $GHE_REMOTE_VERSION)" -ge "$(version 2.16.23)" ]] || \
-         [[ "$GHE_REMOTE_VERSION" =~ 2.17 && "$(version $GHE_REMOTE_VERSION)" -ge "$(version 2.17.14)" ]] || \
-         [[ "$GHE_REMOTE_VERSION" =~ 2.18 && "$(version $GHE_REMOTE_VERSION)" -ge "$(version 2.18.8)" ]] || \
-         [[ "$GHE_REMOTE_VERSION" =~ 2.19 && "$(version $GHE_REMOTE_VERSION)" -ge "$(version 2.19.3)" ]] && \
-         (echo "$line" | grep -q "gist"); then
-        echo "$line"
-      else
-        dirname "$line"
-      fi
-    done
-  }
-
   cat $tempdir/*.rsync | sort | uniq > $tempdir/source_routes
-  (cd $backup_dir/ && find * -mindepth 5 -maxdepth 6 -type d -name \*.git | parse_paths | sort | uniq) > $tempdir/destination_routes
+  (cd $backup_dir/ && find * -mindepth 5 -maxdepth 6 -type d -name \*.git | fix_paths_for_ghe_version | sort | uniq) > $tempdir/destination_routes
```

Comment:
A minor optimisation suggestion:

```suggestion
  (cd $backup_dir/ && find * -mindepth 5 -maxdepth 6 -type d -name \*.git | sort | uniq | fix_paths_for_ghe_version) > $tempdir/destination_routes
```

Reply:
Thanks for the nudge.  You're isn't faster, but this is: `s/sort | uniq/sort -u/`.  If you tell `sort` it can discard all the duplicate keys, then it has considerably less data to work with and can use both less RAM and less CPU.  We may be able to get this step _under_ the 100 seconds it took prior to #524.

Reply:
Eh, maybe just on Mac.  I tried it on Linux and didn't get the same result.

However, another trick that applies here: `uniq | sort | uniq`.  The data we're sorting has sequences of repeated directories any time there are forks in a network.  Running `uniq` as a pre-filter reduces the amount of data that `sort` has to sort.  We do still need the `uniq` at the end, in case the duplicates aren't (fully) clustered in the input.

Reply:
👍 That looks like a great approach, I hadn't considered using `sort` twice like that.